### PR TITLE
Add 'last modified' to S3 object

### PR DESCRIPTION
**Pull request by akyrola**
As title, populates the LastModified metadata into the S3 object. 

The immediate motivation for this stems from enabling finding "latest checkpoint" from a directory of multiple checkpoints. However, surely this can find other use as well.

Testing:

```
(metaflow_dev) MacBook-Pro-5:metaflow akyrola$  aws-vault exec mfdeveloper -- python
Python 3.9.7 | packaged by conda-forge | (default, Sep 14 2021, 01:18:03) 
[Clang 11.1.0 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from metaflow import S3
>>> s3 = S3()
>>> info = s3.info_many(['s3://aapo-mf-test/mnist/t10k-labels-idx1-ubyte.gz', 's3://aapo-mf-test/mnist/train-labels-idx1-ubyte.gz'])
>>> info[0].last_modified, info[1].last_modified
(1632312604.0, 1632312562.0)

>>> o = s3.get('s3://aapo-mf-test/mnist/t10k-labels-idx1-ubyte.gz')
>>> o.last_modified
1632312604.0
>>> oo = s3.get_many(['s3://aapo-mf-test/mnist/t10k-labels-idx1-ubyte.gz', 's3://aapo-mf-test/mnist/train-labels-idx1-ubyte.gz'])
>>> oo[0].last_modified, oo[1].last_modified
(1632312604.0, 1632312562.0)
>>> i = s3.info('s3://aapo-mf-test/mnist/train-labels-idx1-ubyte.gz')
>>> i.last_modified
1632312562.0
>>> m = s3.get('s3://aapo-mf-test/mnist/train-labels-idx1-ubyte.gz', return_info=False)
>>> m.last_modified
>>> 

```

### CODE REVIEW

File: metaflow/datatools/s3.py [hunk 1]
```diff
@@ -956,6 +991,7 @@ def _s3op_with_retries(self, mode, **options):
                         raise MetaflowS3NotFound(err_out)
                     elif ex.returncode == s3op.ERROR_URL_ACCESS_DENIED:
                         raise MetaflowS3AccessDenied(err_out)
+                    print(err_out)
```
[the remark this answers is not in the document]
Comment (akyrola) [reply metaflow/datatools/s3.py:994]: I am suggesting we would leave this, as currently if there is some issue with S3 access (typically an authentication error), metaflow does not give any error but simply keeps retrying (and obviously retries do not even help with authentication issues).

**Comment by akyrola** [comment]
@romain-intel  i have changed the PR to only contain the last_modified change. I believe the context in the PR description is sufficient?

**Review by savingoyal — APPROVED** [review]
LGTM!

**Review by romain-intel — APPROVED** [review]
LGTM.

**Pull request merged**